src/pykeen/pipeline.py

```diff
@@ -242,7 +243,7 @@ class PipelineResult(Result):
     stopper: Optional[Stopper] = None
 
     #: Any additional metadata as a dictionary
-    metadata: Optional[Mapping[str, Any]] = field(default_factory=dict)
+    metadata: Dict[str, Any] = field(default_factory=dict)
```

Comment:
`MutableMapping` instead of `Dict` ? Do we need mutability?

Reply:
Yes mutability is required later

Reply:
Updated annotation in dcad4ed5